Santa Rosa offsite post (#6)
firezone/blog: blog/2023-01-10-first-offsite/index.mdx

## What’s next?

In a few short days, we:
* Aligned on Firezone’s product direction
* Made critical architecture decisions for the gateway and client apps
* Made significant progress on a few key features
* Bonded over great food, conversation, and Mario Kart

We could have accomplished some of these over a few long Zoom calls and chat threads, but not as quickly. There’s something about mind melding over a few days of intense work that’s hard to replicate.

So, will we have another off-site? Definitely!

Until next time!

:::info 👋 thanks for reading
Firezone secures remote access to private resources. This post doesn't have much to do about security, networking, or a fancy new feature. But hey, our product improves the remote work experience - organizing a great off-site helps too.

If you want to try it out, our [Docker install script](https://docs.firezone.dev/) gets Firezone running in just a few minutes.

> :::

Author: @jamilbk added a blurb to the bottom to frame the post + add some info on Firezone. Borrowed from fly.io.

I like how it gives us room to comment and connect things back to FIrezone for these "company building" type posts.

What do you think?

Reviewer: Just needs the `?utm_source=blog`

---
slug: first-offsite-retro
title: Lessons from Firezone’s first offsite
authors: [jason]
tags: [company, offsite]
---

My first startup job was at [Strong](https://www.strong.app/), a workout journal app that was pretty popular at the time. The founders were “digital nomads,” and the team was fully remote.

Back in 2016, this was highly uncommon.

> We spent a lot of time creating processes to work together effectively on opposite sides of the world. One of these rituals was regularly meeting up in person.

Reviewer: ```suggestion
We spent a lot of time creating processes to work together effectively on opposite sides of the world. One of these was meeting up in person regularly.
```

![digital nomads](https://user-images.githubusercontent.com/52545545/211626078-6f64a4f4-ec37-40cd-8bb5-a6b9b122f351.png)

At first, it was by accident. Since everyone was traveling anyway, we would coordinate being in the same city at the same time.

> After working in person we quickly noticed, the way we worked remotely improved. People were more comfortable sharing their ideas, better at giving constructive criticism, and happier in general.

Reviewer: ```suggestion
After working in person, we quickly noticed the way we worked remotely improved. People were more comfortable sharing their ideas, better at giving constructive criticism, and happier in general.
```

> It felt better to work together.

Reviewer: ```suggestion
It just _felt_ better to work together in person.
```

## Why try this at Firezone?

> If you believe a small network security startup can take a sizable bite out of the usual suspects of legacy VPN products, there are a few ways it can happen. Many involve assembling an all-star team who can execute quickly to build a great product.

Reviewer: ```suggestion
If you believe a small network security startup can take a sizable bite out of the usual suspects of legacy remote access products, there are a few ways it can happen. Many involve assembling an all-star team who can execute quickly to build a great product.
```

> If you believe a small network security startup can take a sizable bite out of the usual suspects of legacy VPN products, there are a few ways it can happen. Many involve assembling an all-star team who can execute quickly to build a great product.

Reviewer: ```suggestion
If you believe a small network security startup can take a sizable bite out of the usual suspects of legacy VPN products, there are a few ways it can happen. Many involve assembling an all-star team who can execute quickly to build a great product.
```

I'm not sure I follow this paragraph -- the following two paragraphs don't seem to follow smoothly from it?

This is much easier when everyone is comfortable sharing ideas and engaging in lively debate. Plus, it’s a lot more fun.

So, we decided to try it out. After much planning, canceled bookings, and a few long flights later, our team met up in Santa Rosa, California.

![offsite office](https://user-images.githubusercontent.com/52545545/211517211-ff1d5503-a367-4e6b-a670-067c447ed415.png)

> :::info
> My co-founder Jamil shares similar views about working together in person. He’s an even earlier pioneer in working remotely. I keep bugging him to write a blog about his 3 year journey living in a tiny house towed by his truck.
> 
> For now, I’ll leave you with a link to his [slideshow](https://tinyhouse.jamilbk.com/).
> :::
> 

Reviewer: Hmm, maybe this could link to a future post about my experience, instead of the linking to slideshow, suggest removing for now.

```suggestion
```

## What did we work on?

Our itinerary included architecture planning, a multi-day hackathon, and plenty of fun activities. Let’s start with what we built.

### REST API v0

One common compliment from our users is how snappy and effortless it is to manage Firezone with the Web UI. Many IT teams support hundreds of concurrent users without a hitch. However, as fetching as our web portal is, we’d prefer if you rarely had to open it up. 

> Our REST API follows a series of updates to make Firezone easier to automate. This is why we made Docker the preferred (and soon-to-be only) method of deploying Firezone. And also why we’re building on the work from our community to release a first-party Terraform provider.

Reviewer: ```suggestion
Our [REST API](https://docs.firezone.dev/reference/rest-api/?utm_source=blog) follows a series of updates to make Firezone easier to automate. This is why we made Docker the preferred method of deploying Firezone, and also why we’re building on the work from our community to release a first-party Terraform provider.
```

Will announce the details in 0.7 blogpost, can link to that from here when ready.

Take a look at the work in progress in [this PR](https://github.com/firezone/firezone/pull/1205). 

### Documentation improvements

We added additional docs on a few topics, including debug logs, split tunnel DNS considerations, and common troubleshooting issues.

SEO was not top of mind when writing our docs, but we found ourselves part of the zero-sum game that appeases our Google overlords.

Whether for goodwill or profit, you’ll find guides for Firezone of varying quality on the internet. Recently, we’ve noticed these ranking higher than our first-party docs.

![google results](https://user-images.githubusercontent.com/52545545/211517741-4efb596a-3765-40d8-b483-f807f3a88e86.png)

This is a problem because these guides are quickly outdated and often inaccurate. We know many prefer searching on Google instead of directly visiting our docs, so we made some updates to rank higher for popular searches.

Take a look at the PR [here](https://github.com/firezone/firezone/pull/1194).

> 

Reviewer: This section feels more like a development / product update, and I fear it would be confused as an announcement of sorts. Think it might be possible to discuss "what we worked on" more generally? Could summarize this section.

Author: It does seem like a better fit for a general post on remote work to not focus too much on specifics only our user base cares about. I'll summarize and we can link other posts from this section once we write them (e.g. 0.7.0 launch).

Back in 2016, this was highly uncommon.

We spent a lot of time creating processes to work together effectively on opposite sides of the world. One of these was meeting up in person regularly.

![digital nomads](https://user-images.githubusercontent.com/52545545/211626078-6f64a4f4-ec37-40cd-8bb5-a6b9b122f351.png)

At first, it was by accident. Since everyone was traveling anyway, we would coordinate being in the same city at the same time.

After working in person, we quickly noticed the way we worked remotely improved. People were more comfortable sharing their ideas, better at giving constructive criticism, and happier in general.

It just _felt_ better to work together in person.

## Why try this at Firezone?

If you believe Firezone can take a sizable bite out of the usual suspects of legacy remote access products, we'll need to use every advantage we have.

As a startup, our biggest advantage is the ability to move quickly as a team.

This is much easier when everyone is comfortable sharing ideas and engaging in lively debate. Plus, it’s a lot more fun.

> So, we decided to try it out. After much planning, canceled bookings, and a few long flights later, our team met up in Santa Rosa, California.

Reviewer: ```suggestion
So we decided to try it out. After much planning, canceled bookings, and a few long flights later, our team met up in Santa Rosa, California.
```

![offsite office](https://user-images.githubusercontent.com/52545545/211517211-ff1d5503-a367-4e6b-a670-067c447ed415.png)

## What did we do?

Our itinerary included architecture planning, a multi-day hackathon, and plenty of fun activities around the Bay Area. 

The focus of the off-site was building. We each defined a few tasks that could realistically be completed by the end of the trip. Here's a short summary of what we accomplished:

* REST API v0: Many IT teams use our snappy web portal to support hundreds of concurrent users without a hitch. However, we’d prefer if you rarely had to open it up. [PR here](https://github.com/firezone/firezone/pull/1155).
* Updating the Gateway: The Firezone gateway is tightly coupled with the control plane. To support larger teams with multi-site environments, we’ll need to support managing multiple gateways under a single control-plane. Work-in-progress in [this PR](https://github.com/firezone/firezone/pull/1205). 

> * Docs improvements: We wrote a few new docs on debug logs, split tunnel DNS considerations, and common troubleshooting issues. Details in [this PR](https://github.com/firezone/firezone/pull/1194).

Reviewer: Probably shouldn't link to features/PRs that aren't shipped yet, might confuse folks.

```suggestion
The focus of the off-site was building. We each defined a few tasks that could realistically be completed by the end of the trip. So what did we work on?

* A new Firezone gateway (coming in 0.8!) that will eventually allow decoupling the web portal from the WireGuard data plane. A single Firezone web portal will soon be able to manage hundreds or even thousands of little Firezone gateways securing access for their respective private resources. More details to come!
* A [new REST API in 0.7](https://github.com/firezone/firezone/releases) to allow configuring Firezone in a more automatable way. Users love our snappy web portal, but nothing beats the power of an API for automation.
* Various documentation improvements to improve consistency between Omnibus and Docker sections, detail logging, and implement a v1 style guide.
```

> Of course, a visit to the Bay Area would not be complete without a tour of San Francisco. We visited the usual landmarks and had a lot of seafood.

Reviewer: ```suggestion
Of course, no visit to the Bay Area would be complete without a tour of San Francisco. It rained most of the time, so we missed out on much of the wonderful hiking the Bay has to offer. But we still managed to snag a few In-N-Out burgers and catch beautiful glimpses of the Golden Gate bridge.
```

Author: rewrote a bit in my own words

>Of course, no visit to the Bay Area would be complete without a tour of San Francisco. It rained most of the time, so we missed out on much of the picturesque hiking. But we still managed to snap some photos of the Golden Gate bridge and devour a few In-N-Out burgers.

> Of special mention are the team dinners. As someone whose culinary talents rival his passion for building security products, Jamil (Co-founder & CEO) made our team dinners very memorable. You can follow his musings at [@jamilbk](https://twitter.com/jamilbk).

Reviewer: Added a little more pizazz to fill this section out more -- I think it's one of the meatier takeaways of the post (no pun intended). lmk thoughts

```suggestion
Of special mention are the team dinners. Remote companies lack that familiar bond typically formed over team lunches in the office, so we sought to recreate that when we met in person. As an added bonus, we also split cooking duties. In many ways preparing food for your teammates is like opening a pull request: make something, request feedback on it, then merge and enjoy. As someone whose culinary talents rival his passion for building security products, our resident chef [Jamil](https://twitter.com/jamilbk) led most of the culinary endeavors, including a 2-hour drive around Santa Rosa in search of live dungeness crab.

![A56C8756-B231-4351-BA0E-3C37C8060957_1_105_c](https://user-images.githubusercontent.com/167144/214394961-ecbff7d8-2f9c-48a6-a906-c387366169f3.jpeg)
```

Author: Love it! Included in my commit.

![collage](https://user-images.githubusercontent.com/52545545/211988059-02f3083a-116b-462e-a5ff-90af4fab63fc.png)

> If this looks like something you’d be interested in doing, [we’re always hiring](https://www.ycombinator.com/companies/firezone/jobs)!

Reviewer: ```suggestion
```

## Lessons learned for next time

> Since this was our first offsite, some things could have gone better. Here’s a small list of hiccups and our learnings.

Reviewer: ```suggestion
Since this was our first offsite, some things could have gone better. Here’s a small list of hiccups and other things we'd do differently next time around.
```

> ### Close your tabs!

Reviewer: ```suggestion
### Stay connected!
```

Reviewer: Is `Close your tabs` a typo here? Just a little confused what it has to do with the section below, hence the suggestion.

Author: Oh, it was supposed to ref chrome eating up all the battery. Maybe something more obvious and related...

- Turn on low power mode!
- Switch to vim
- Laptops on life support

<img src="https://user-images.githubusercontent.com/52545545/211503302-4b2e2e74-06e8-4929-aa5b-a336af25a450.png" width="400" />

> If you’re booking an Airbnb, check for planned maintenance and outages. We were notified three days before arrival the power would be out during work hours on two weekdays.

Reviewer: ```suggestion
If you’re booking an Airbnb, check for planned maintenance and outages! We were notified three days before arrival the power would be out during two full workdays.
```

> We made it work by sharing battery banks and migrating to the local coffee shop when our batteries died. Luckily Firezone is just about peak power efficiency for a VPN, so we all gained a few extra minutes of battery.

Reviewer: ```suggestion
Luckily a few of us had large battery banks and unlimited hotspots to tide us over for most of the outage. When those died, we migrated to our local coffee shop where power and WiFi were surprisingly abundant. _Editor's note: Coffee shops in Santa Rosa are much more accommodating as a makeshift office than most in San Francisco._
```

> When you're building software, electricity and internet are pretty much the only things you can't do without. Next time we'll pack a large power bank.

Reviewer: ```suggestion
When you're building software, electricity and internet are vital -- splurge for that bigger battery bank and faster hotspot!
```

> ### Lumbar support is under-rated

Reviewer: ```suggestion
### Lumbar support is underrated
```

You’ll sacrifice some ergonomics when you work out of an Airbnb. Put a premium on large work surfaces, open common areas, and comfortable chairs. We learned the hard way after a few days working hunched over on a sofa.

Even better, leave heads down work until after the offsite. Instead, prioritize architecture discussions, brainstorming sessions, and design sprints. 

### Stock the right fuel

We got carried away at Costco. Our out-of-town guests had never witnessed the spectacle of an American snack aisle. Fortunately, our talented engineers can turn Funyuns into elegant lines of Rust code (shoutout [@Gabi](https://github.com/conectado)). 

All jokes aside, you’ll need the right fuel for your hackathons and deep product discussions. We recommend stocking up with plenty of coffee and healthy snacks.

> ### Plan ahead, it will pay off

Reviewer: ```suggestion
### Plan ahead: it'll pay off
```

As is often the case with startups, we planned the trip on short notice. This meant many last-minute restaurant bookings and trips to the store for basic supplies.

So, make a shopping list, book a few restaurants in advance, and have a backup plan. If you're looking for a template, take a look at this [guide](https://posthog.com/handbook/company/offsites#how-to-plan-an-offsite-in-8-weeks---a-checklist) from our friends at PostHog. A great checklist is hard to beat.

## What’s next?

In a few short days, we:
* Aligned on Firezone’s product direction
* Made critical architecture decisions for the gateway and client apps
* Made significant progress on a few key features
* Bonded over great food, conversation, and Mario Kart

We could have accomplished some of these over a few long Zoom calls and chat threads, but not as quickly. There’s something about mind melding over a few days of intense work that’s hard to replicate.

So, will we have another offsite? Definitely!

> You can also expect more blog posts from us. One of our New Year’s resolutions at Firezone is to write more. We want to be transparent with how we build our product, run our company, and make important decisions that impact our users. We do this because we don’t take your trust in Firezone for granted.

Reviewer: This seems a bit off-topic, and may be better saved for a formal post on transparency.

```suggestion
```
